packages/common/src/compiler/exploreCompiler.test.ts

```diff
@@ -139,12 +140,14 @@ describe('Explores with a base table and joined table', () => {
             compiler.compileExplore(joinedExploreWithSubsetOfFields),
         ).toStrictEqual(compiledJoinedExploreWithSubsetOfFields);
     });
-    test('should throw error if referenced field is removed from join', () => {
-        expect(() =>
+    test('should compile with a subset of fields selected on join what dont include the fields in the join SQL', () => {
```

Comment:
We already had a test for this scenario but we were forcing the user to include all referenced fields in the fields array. Updated mock+test to reflect the new behaviour.